packager/mpd/base/representation.cc

```diff
@@ -328,10 +328,17 @@ bool Representation::HasRequiredMediaInfoFields() const {
   return true;
 }
 
-void Representation::AddSegmentInfo(int64_t start_time, int64_t duration) {
+void Representation::AddSegmentInfo(int64_t start_time, int64_t duration, uint64_t segment_index) {
   const uint64_t kNoRepeat = 0;
   const int64_t adjusted_duration = AdjustDuration(duration);
 
+    if (segment_infos_.empty()) {
+    if (mpd_options_.mpd_params.target_segment_duration > 0) {
+      // Store segment index as 1 based.
+      start_number_ = segment_index + 1; 
+    }
+  }
```

Comment:
It seems that segment_index or start_segment_index should be added as a member of SegmentInfo. Then we can remove |start_number_|.

Otherwise the code at line 462 is incorrect.

Reply:
Thanks for the review. Let me try this out. Thinking about how copy constructor changes representation will affect the unit test (RepresentationClone).

Reply:
@kqyang For representation_unittest.cc::TEST_P(TimeShiftBufferDepthTest, Generic) as the segment duration changes, segment index is 34 instead of 1002 which it would have been because of increasing |start_number_|. How this test be changed changed in order to rely on start_segment_index which is stored in SegmentInfo?

Reply:
@kqyang any thoughts?

Reply:
Sorry for the late reply. I think it actually surfaces a problem in the unit-test. I think it is better to have your logic here in the test :).

We can define a member variable in SegmentTemplateTest called: start_segment_index_. It is initialized to start_time/duration in AddSegments if not initialized; increment by 1 in for-loop.

```c++
 void AddSegments(int64_t start_time,
                   int64_t duration,
                   uint64_t size,
                   int repeat) {
    ...

    if (start_segment_index == -1) 
     start_segment_index = start_time / duration;

    for (int i = 0; i < repeat + 1; ++i) {
      representation_->AddNewSegment(start_time, duration, size, start_segment_index++);
      start_time += duration;
      bandwidth_estimator_.AddBlock(
          size, static_cast<double>(duration) / kDefaultTimeScale);
    }
  }
```

Reply:
I have added this change.